app/experimenter/nimbus-ui/src/components/AppLayoutSidebarLocked/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import { RouteComponentProps, useParams } from "@reach/router";
+import Container from "react-bootstrap/Container";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Nav from "react-bootstrap/Nav";
+import Scrollspy from "react-scrollspy";
+import { AnalysisData } from "../../lib/visualization/types";
+import { analysisAvailable } from "../../lib/visualization/utils";
+import { StatusCheck } from "../../lib/experiment";
+import { DisabledItem } from "../DisabledItem";
+import { LinkNav } from "../LinkNav";
+import { ReactComponent as ChevronLeft } from "../../images/chevron-left.svg";
+import { ReactComponent as Clipboard } from "../../images/clipboard.svg";
+import { ReactComponent as BarChart } from "./bar-chart.svg";
+import LinkExternal from "../LinkExternal";
+
+export const RESULTS_LOADING_TEXT = "Checking results availability...";
+
+const getSidebarItems = (metrics: { [metric: string]: string }) => {
+  const keys: string[] = [];
+  const labels: string[] = [];
+
+  Object.keys(metrics!).map((metric) => {
+    keys.push(metric);
+    labels.push(metrics[metric]);
+  });
+
+  return { keys, labels };
+};
+
+type AppLayoutSidebarLockedProps = {
+  testid?: string;
+  children: React.ReactNode;
+  status?: StatusCheck;
+  analysis?: AnalysisData;
+  analysisLoadingInSidebar?: boolean;
```

Comment:
I believe we can remove the `?` on these since they're required. I think `analysisError` returns `undefined` if there's not an error so that one can probably stay as "Error or undefined".